receiver/googlecloudspannerreceiver/cardinality.md

```diff
@@ -0,0 +1,75 @@
+# Cloud Spanner: Controlling Metrics Cardinality for Open Telemetry Receiver
+
+## Overview
+This document describes the algorithm used to control the cardinality of the Top N metrics exported by Cloud Spanner receiver in OpenTelemetry collector.
+With cardinality limits enforced in Cloud Monitoring and possibly in Prometheus, it is critical to control the cardinality of the metrics to avoid drops in time series which can limit the usability of Cloud Spanner custom metrics.
+
+## Background
+Cloud Monitoring has 200,000 active time series (streams) [limits](https://cloud.google.com/monitoring/quotas) for custom metrics per project.
+Active time series translates to streams that are sent in the last 24 hours.
+This means that the number of streams that can be sent by OpenTelemetry collector should be controlled to this limit as the metrics will be dropped by Cloud Monitoring.
+
+## Calculation
+For simplicity purpose and for explaining of calculation, which is currently done in implementation, lets assume that we have 1 project, 1 instance and 1 database.
+Such calculations are done when collector starts. If cardinality total limit is different(for calculations below we assume it is 200000) - numbers will be different.
+
+According to the metrics metadata configuration(at the moment of writing this document) we have:
+- 30 low cardinality metrics(amount of total N + active queries summary metrics);
+- 26 high cardinality metrics(amount of top N metrics).
+
+For low cardinality metrics allowed amount of time series is **projects x instances x databases** and in our case is **1 x 1 x 1 = 1**.
+
+Then, remaining quota of allowed time series in 24 hours is **200000 - 30 = 199970**.
+
+In 24 hours we can have max **24 x 60 = 1440** time series values.
+
+So, for high cardinality metrics we have **199970 / 26 = 7691** allowed time series per metric.
+
+This means each metric converted from these columns should not have a cardinality of 7691 per day.
+
+Taking into account 1440 minutes we'll obtain **7691 / 1440 = 5** new time series per minute.
```

Comment:
It should be 1440 time series, sorry. This value is our limit by timestamp (per minute). In case this is reached we are not exporting new time series, only those which are in cash. This is the intent of algorithm.